pennylane_lightning/src/lightning_qubit.cpp

```diff
@@ -99,9 +83,132 @@ VectorXcd apply (
     }
 }
 
+vector<int> calculate_tensor_indices(const vector<int> &wires, const vector<int> &tensor_indices) {
```

Comment:
These function definitions were moved here from the `.hpp` file. Docstrings were left in `.hpp` with the declaration of each function.